libdevice/fallback-cassert.cpp

```diff
@@ -27,6 +58,9 @@ DEVICE_EXTERN_C void __devicelib_assert_fail(const char *expr, const char *file,
                      // (func) ? func : "<unknown function>",
                      func, gid0, gid1, gid2, lid0, lid1, lid2, expr);
 
+  // FIXME uncomment the following line after clang changes
+  // cl::sycl::detail::AssertHappenedMem.Flag = 1;
```

Comment:
What is the semantics of this flag setting in the case of two kernels being executed concurrently?
Where is the initialization/reset of this flag happening?

Reply:
Changing of the flag should be atomic operation.
It should look like this:
```c++
int Expected = 0;
int Desired = 1;

CAS(&AssertHappenedMem.Flag, Expected, Desired);
```

Reply:
So the flag is set to `true` whenever assertion is triggered by any kernel, and it remains `true` until the program completion.  In other words, querying `__devicelib_assert_read` after some kernel invocation does not mean that this particular kernel triggered an assertion.  Can you please add this into the documentation?

Reply:
Updated document.